rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/ProductMilestoneEndpoint.java

```diff
@@ -518,4 +520,36 @@ List<DeliveredArtifactInMilestones> compareArtifactVersionsDeliveredInMilestones
     Graph<ProductMilestone> getMilestonesSharingDeliveredArtifactsGraph(
             @Parameter(description = PM_ID) @PathParam("id") String milestoneId,
             @QueryParam("depthLimit") @Min(0) @Max(5) @DefaultValue("5") Integer depthLimit);
+
+    static final String GET_DELIVERED_ARTIFACTS_SHARED_IN_MILESTONES = "Fetches Artifacts delivered in both of the Milestones.";
+
+    /**
+     * {@value GET_DELIVERED_ARTIFACTS_SHARED_IN_MILESTONES}
+     *
+     * @param pageParameters
+     * @param milestone1Id
+     * @param milestone2Id
+     * @return
+     */
+    @Operation(
+            summary = GET_DELIVERED_ARTIFACTS_SHARED_IN_MILESTONES,
+            responses = {
+                    @ApiResponse(
+                            responseCode = SUCCESS_CODE,
+                            description = SUCCESS_DESCRIPTION,
+                            content = @Content(schema = @Schema(implementation = SwaggerPages.ArtifactPage.class))),
+                    @ApiResponse(
+                            responseCode = INVALID_CODE,
+                            description = INVALID_DESCRIPTION,
+                            content = @Content(schema = @Schema(implementation = ErrorResponse.class))),
+                    @ApiResponse(
+                            responseCode = SERVER_ERROR_CODE,
+                            description = SERVER_ERROR_DESCRIPTION,
+                            content = @Content(schema = @Schema(implementation = ErrorResponse.class))) })
+    @GET
+    @Path("/delivered-artifacts/shared")
+    Page<Artifact> getDeliveredArtifactsSharedInMilestones(
+            @Valid @BeanParam PageParameters pageParameters,
+            @Parameter(description = PM_ID) @QueryParam("milestone1") String milestone1Id,
```

Comment:
I'd validation with something akin to `@NotBlank` as it's a required parameter right? Same for the second milestone ID.

Reply:
`@NotBlank` looks about right